app/src/main/java/com/example/coinage/fragments/AddTransactionFragment.java

```diff
@@ -98,15 +107,38 @@ public void onDateSet(DatePicker view, int year, int month, int day) {
                 etDate.setText(dateFormat.format(myCalendar.getTime()));
             }
         };
-        etDate.setOnClickListener((View v) ->
-                new DatePickerDialog(getContext(),datePicker,myCalendar.get(Calendar.YEAR),myCalendar.get(Calendar.MONTH),myCalendar.get(Calendar.DAY_OF_MONTH)).show());
+
+        etDate.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if(hasFocus) {
+                    etDate.setInputType(InputType.TYPE_NULL);
+                    new DatePickerDialog(getContext(),datePicker,myCalendar.get(Calendar.YEAR),myCalendar.get(Calendar.MONTH),myCalendar.get(Calendar.DAY_OF_MONTH)).show();
+                }
+            }
+        });
+
+        // category spinner
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(getContext(),
+                R.array.categoriesAdd,
+                R.layout.custom_spinner_item);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        tiCategory.setAdapter(adapter);
+        tiCategory.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
```

Comment:
why is this `onItemSelectedListener` being set with empty implementations for both methods?

Reply:
I don't actually need the functions but they're required?

Reply:
Oh wait, for some reason I thought it was a required function for spinners, I should've just deleted it lol